### DISCUSSION TITLE
How to apply a specific pattern to the whole document via Hotkeys (command palette)

### DISCUSSION THREAD

**Discussion by e-alizadeh** [jglev/obsidian-apply-patterns-plugin#48] [Q&A] [answered]
Hi again, 

I've created a pattern named "Replace abc with def" and I want to apply that to a whole document via command palette (in particular via Hotkey). 

Current, I have to call "Apply Patterns: Apply pattern to whole document" and then select the pattern. However, my goal is to have a hotkey to run the pattern to the whole document. 

Thanks in advance

**Comment by FelipeRearden** [comment]
In the bottom of the Settings, we have a the Commands Section. 

In the Commands section you can define one or more Patterns to a specific Command. 
- Moreover, you can define the way that the Pattern will be applied (in your case, `Whole Document`

That's the path to trigger a pattern from a hotkey using Commands. 

I hope it helps 🙏

**Reply by e-alizadeh** [reply]
Thanks for your response. 
I tried different options in Command section but with no success. 
Nothing is added or filtered in the Command section. 
Each time, I closed the Obsidian and reopened it just to reload all configurations.

**Reply by jglev** [reply, marked as answer]
@FelipeRearden, thank you for explaining!

@e-alizadeh, I'm sorry to hear you've been having trouble!

If your pattern is named "Replace abc with def", then (1) copying the text below to your clipboard, (2) going into the Apply Patterns settings, (3) Clicking "Import" under the Commands section at the bottom of the settings page, and (4) reloading / re-starting Obsidian _should_ add a command to the Command Palette. In addition, **the Pattern itself must have at least one Rule within it to show up in the Command Palette.**

To confirm, does this work for you?

The text to copy:

```
[
  {
    "name": "Replace abc with def",
    "patternFilter": "Replace abc with def",
    "selection": false,
    "lines": false,
    "document": true,
    "clipboard": false,
    "clipboardLines": false
  }
]
```

**Reply by e-alizadeh** [reply]
Thanks a lot @jglev. It's working now 😄 

Above instructions were very clear. I believe others would benefit from it as well if it's in README.
